src/hotspot/share/classfile/classListParser.cpp

```diff
@@ -398,30 +453,27 @@ InstanceKlass* ClassListParser::load_class_from_source(Symbol* class_name, TRAPS
   if (strncmp(_class_name, "java/", 5) == 0) {
     log_info(cds)("Prohibited package for non-bootstrap classes: %s.class from %s",
           _class_name, _source);
-    return NULL;
+    THROW_NULL(vmSymbols::java_lang_ClassNotFoundException());
```

Comment:
Is that the right exception to use here? I'm not clear of the context for encountering this. What does regular classloading do in such a case?

Reply:
Same as above. This will be improved in [JDK-8263469](https://bugs.openjdk.java.net/browse/JDK-8263469).